### ISSUE TITLE
[🐛] "Usage of "messaging().registerDeviceForRemoteMessages()" is not required..." warning is always triggered on iOS Simulator

### ISSUE THREAD

**Issue by marcfrankel** [invertase/react-native-firebase#6501]
## Issue

We have a fairly privacy and GDPR-focused app so we try and turn off as much of the Firebase auto connecting as possible. I noticed a small bug where the warning 
`
Usage of "messaging().registerDeviceForRemoteMessages()" is not required. You only need to register if auto-registration is disabled in your 'firebase.json' configuration file via the 'messaging_ios_auto_register_for_remote_messages' property.
`

Will trigger even when that flag is set to false in your Firebase.json if you're running on an iOS simulator. My guess is it has something to do with the following check where this error message is logged:

**@react-native-firebase/messaging/lib/index.js line 271**
```js
const autoRegister = this.firebaseJson['messaging_ios_auto_register_for_remote_messages'];
    if (autoRegister === undefined || autoRegister === true) {
      // eslint-disable-next-line no-console
      console.warn(
        `Usage of "messaging().registerDeviceForRemoteMessages()" is not required. You only need to register if auto-registration is disabled in your 'firebase.json' configuration file via the 'messaging_ios_auto_register_for_remote_messages' property.`,
      );
    }
```
But I'm honestly not sure. The exact code, setup, and build works fine on a physical iPhone.

Also, apologies for the barely filled out bug report but I figured most was not relevant. If you think any more info would be helpful I'm happy to further fill it out.

Appreciate all your work!

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
# N/A
```

#### `firebase.json` for react-native-firebase v6:

```json
{
    "$schema": "./node_modules/@react-native-firebase/app/firebase-schema.json",
    "react-native": {
        "analytics_auto_collection_enabled": false,
        "google_analytics_automatic_screen_reporting_enabled": false,
        "analytics_default_allow_ad_personalization_signals": false,
        "app_data_collection_default_enabled": false,
        "google_analytics_adid_collection_enabled": false,
        "crashlytics_auto_collection_enabled": false,
        "perf_auto_collection_enabled": false,
        "crashlytics_debug_enabled": false,
        "crashlytics_javascript_exception_handler_chaining_enabled": false,
        "in_app_messaging_auto_collection_enabled": false,
        "messaging_ios_auto_register_for_remote_messages": false,
        "messaging_auto_init_enabled": false,
        "android_task_executor_maximum_pool_size": 10,
        "android_task_executor_keep_alive_seconds": 3,
        "analytics_idfv_collection_enabled": false
    }
}
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
 OUTPUT GOES HERE
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - 14.11.1
- **`Firebase` module(s) you're using that has the issue:**
  -  cloud messaging (messages)
- **Are you using `TypeScript`?**
  - Y

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
Strange - so if I read this correctly this is a simulator / real device difference? You confirm that when you run on a real device with the same firebase.json (after a pod install and build on each) that simulator prints this but real device does not?

That's unexpected for sure but I don't use this flag personally so I may have missed it.

Relevant data might be Xcode version, runtime specifics of simulator and real device

**Comment by marcfrankel** [comment]
Yeah, it's the same build on both the simulator and physical device. The other relevant information might be that I'm using Expo and the builds are performed through EAS. I doubt it has much to do with this, but I figured I'd mention it since Xcode version on the physical build is a tad out of my control. However, in terms of the simulator, it's running 15.2 with xcode version  13.4.1

**Comment by mikehardy** [comment]
If I understand correctly and this is not affecting production builds, implying that your user's expectations of privacy are not degraded, then this won't be as a high priority.

But it's still puzzling. I'm trying to think under what conditions you might have a firebase.json with those contents and still hit that code branch, and I can't think of one. That implies that perhaps the firebase.json processing is totally failing somehow 

These lines, where the firebase.json is included as a base64 string in Info.plist for runtime availability: 

https://github.com/invertase/react-native-firebase/blob/a05f63d5cb208d9b6f645a0743584f58a942d6bc/packages/app/ios_config.sh#L77-L88

But there was a problem there recently (#6194) which I fixed and turned into a total build failure (#6234) so it should not be able to slip past there without your firebase.json being ingested in the plist and then loaded into app at runtime

Now we've got more pertinent information missing that the template would have provided ;-) - are you on a version that includes that change?

14.9.2 or higher https://github.com/invertase/react-native-firebase/commit/1f609d379117532d014bc44735827d38d79a36e9
